### ISSUE TITLE
missing findOrCreateInvite

### ISSUE THREAD

**Issue by qkral** [jadolg/rocketchat_API#86]
I would need to use the invite methods described here: https://docs.rocket.chat/api/rest-api#invites in particular the findOrCreateInvite that is described here in the REST API https://docs.rocket.chat/api/rest-api/methods/invites/findorcreateinvite. I don't see it in your python API. Could it be implemented?

**Comment by qkral** [comment]
when I use your API in my django project I randomly get an error message sometimes saying:


Exception Type: | RocketConnectionException
/Users/quenti/anaconda3/envs/conf/lib/python3.8/site-packages/rocketchat_API/rocketchat.py in login, line 108
/Users/quenti/anaconda3/envs/conf/bin/python3

rocket = RocketChat('website', settings.SECRETROCK, server_url='https://chat.yyyy.com',session=session) …
▶ Local vars
/Users/quenti/anaconda3/envs/conf/lib/python3.8/site-packages/rocketchat_API/rocketchat.py in __init__
            self.login(user, password)  # skipcq: PTC-W1006 …
▶ Local vars
/Users/quenti/anaconda3/envs/conf/lib/python3.8/site-packages/rocketchat_API/rocketchat.py in login
            raise RocketConnectionException() 

Basically it seems that sometimes the connection to my chat doesn't work but I'm always using the same identifiers so it's kind of random. Would you know why it happens? Could it be because I've asked for another connection 1s before (which happens) and somehow it doesn't allow too many demands at once or something to do with session=session that should be closed?

Thanks,
Quentin

**Comment by jadolg** [comment]
I'll try to implement the `findOrCreateInvite` as soon as possible. As for the error you are mentioning, how often are you executing it? I have had problems before because by default Rocket.Chat uses rate-limiting in their API. This was once a problem while running the tests and might also be what's going on.

**Comment by qkral** [comment]
Thank you for findOrCreateInvite! I run the API every time I load a page and sometimes on the new page I have just loaded, I have to run the API one more time consecutively and this is indeed when I have troubles (but not always). I had to set a timer of 20s when I get the error and then I wait for 20s and then it works... Any way to change the API rate limit?

**Comment by qkral** [comment]
Thanks I found the rate limiter options in the admin panel of the rocket chat and it indeed seems to solve the problem!

**Comment by jadolg** [comment]
Hi @qkral 
I'm working on it but looks like there's some problem on the API side.
I have drafted a PR https://github.com/jadolg/rocketchat_API/pull/88 but tests are not passing.
You can follow the discussion here https://open.rocket.chat/channel/python_rocketchat_api?msg=nEzzMv4AdP8QTap2b

**Comment by jadolg** [comment]
the latest version (1.7.0) includes the change already.

**Comment by qkral** [comment]
Thank you so much!

**Comment by qkral** [comment]
I have just tried and it seems that the link it creates doesn't send to the room I put in rid but rather to /home every time. Have you tried the links? Does it work for you for private groups? If I compare the link I can make in the admin panel for the room I want to invite people to (https://go.rocket.chat/invite?host=chat.xxxx.com&path=invite%2FbG3Nub
) with the links the findOrCreateInvite makes (https://go.rocket.chat/room?host=chat.xxxx.com&rid=TAe9arKePs45vavGS), they don't look alike. Is there an issue I don't understand here?
